public/app/plugins/datasource/prometheus/components/PromQueryEditor.tsx

```diff
@@ -37,7 +37,9 @@ export class PromQueryEditor extends PureComponent<Props, State> {
 
   constructor(props: Props) {
     super(props);
-    const { query } = props;
+    // Use default query to prevent undefined input values
+    const defaultQuery: PromQuery = { refId: '', expr: '', legendFormat: '', interval: '' };
```

Comment:
I think `refId` is not required here - it should be passed into the query editor by the metric panel editor. So `refId` is always set here. Maybe 
```suggestion
    const defaultQuery: Partial<PromQuery> = { expr: '', legendFormat: '', interval: '' };
```
will be better in this case.

Reply:
Ah, nice, didn't know about `Partial`. Typescript was complaining about missing `refId` prop so that's why I added it.

Reply:
Yea, `refId` is required prop, but it's a part of the type `PromQuery` extends. So `Partial<>` works fine in this case.